### DISCUSSION TITLE
请问有检测语音是否是普通话的模型么

### DISCUSSION THREAD

**Discussion by BrightXiaoHan** [PaddlePaddle/PaddleSpeech#2864] [Q&A]
请问有检测语音是否是普通话的模型么，paddlespeech中有相关的模型或者实现么

**Comment by yt605155624** [comment]
没有，可以自己收集语料跑一个语音分类的模型试试

**Reply by BrightXiaoHan** [reply]
从头训还是找个模型finetune比较好一点？

**Comment by zxcd** [comment]
你是需要检测单句是否是英文或者普通话这种情况吗？

**Reply by BrightXiaoHan** [reply]
检测普通话和粤语

**Reply by zxcd** [reply]
目前没有单独检测这两种话的模型，如果精度要求比较高建议自己训练一个分类模型。

如果不存在单句中包含两种语言的情况，且对精度要求不是特别高的情况下，可以使用尝试调用whisper模型。whisper目前包含语种检测功能，由于没有集成粤语检测，对于你的情况应该会把粤语识别成某个小语种，普通话仍然识别为普通话。

具体精度问题你可以针对自己的数据跑CLI尝试一下，使用参考 https://github.com/PaddlePaddle/PaddleSpeech/tree/develop/demos/whisper

希望能对你有所帮助 : )

**Reply by BrightXiaoHan** [reply]
感谢
